app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -42,20 +39,45 @@ class SearchRepositoriesViewModel(private val repository: GithubRepository) : Vi
     @Volatile
     private var lastQueryValue: String? = null
     @Volatile
-    private var lastSearchResult: Flow<PagingData<Repo>>? = null
+    private var lastSearchResult: Flow<PagingData<UiModel>>? = null
 
     /**
      * Search a repository based on a query string.
      */
-    fun searchRepo(queryString: String): Flow<PagingData<Repo>> {
+    fun searchRepo(queryString: String): Flow<PagingData<UiModel>> {
         val result = lastSearchResult
-        if(queryString == lastQueryValue && result != null){
+        if (queryString == lastQueryValue && result != null) {
             return result
         }
         lastQueryValue = queryString
-        val newResult = repository.getSearchResultStream(queryString)
+        val newResult: Flow<PagingData<UiModel>> = repository.getSearchResultStream(queryString)
+                .map { pagingData -> pagingData.map { UiModel.RepoItem(it) as UiModel } }
```

Comment:
if you skip this map, then the subsequent map shouldnt ahve to do the is checks - they'll always be true

Reply:
Disregard this and the below comment for now!

Realized this is a bug in the generics of this function that prevents you from doing this more cleanly: https://issuetracker.google.com/issues/151050495